test/google/gax/util_test.rb

```diff
@@ -0,0 +1,200 @@
+require 'stringio'
+
+describe Google::Gax do
+  describe '#to_proto' do
+    REQUEST_NAME = 'path/to/ernest'.freeze
+    USER_NAME = 'Ernest'.freeze
+    USER_TYPE = :ADMINISTRATOR
+    POST_TEXT = 'This is a test post.'.freeze
+    MAP = {
+      'key1' => 'val1',
+      'key2' => 'val2'
+    }.freeze
+
+    it 'creates a protobuf message from a simple hash' do
+      hash = { name: USER_NAME, type: USER_TYPE }
+      user = Google::Gax.to_proto(hash, Google::Protobuf::User)
+      _(user).must_be_kind_of(Google::Protobuf::User)
+      _(user.name).must_equal(USER_NAME)
+      _(user.type).must_equal(USER_TYPE)
+    end
+
+    it 'creates a protobuf message from a hash with a nested message' do
+      request_hash = {
+        name: REQUEST_NAME,
+        user: Google::Protobuf::User.new(name: USER_NAME, type: USER_TYPE)
+      }
+      request = Google::Gax.to_proto(request_hash, Google::Protobuf::Request)
+      _(request).must_be_kind_of(Google::Protobuf::Request)
+      _(request.name).must_equal(REQUEST_NAME)
+      _(request.user).must_be_kind_of(Google::Protobuf::User)
+      _(request.user.name).must_equal(USER_NAME)
+      _(request.user.type).must_equal(USER_TYPE)
+    end
+
+    it 'creates a protobuf message from a hash with a nested hash' do
+      request_hash = {
+        name: REQUEST_NAME,
+        user: { name: USER_NAME, type: USER_TYPE }
+      }
+      request = Google::Gax.to_proto(request_hash, Google::Protobuf::Request)
+      _(request).must_be_kind_of(Google::Protobuf::Request)
+      _(request.name).must_equal(REQUEST_NAME)
+      _(request.user).must_be_kind_of(Google::Protobuf::User)
+      _(request.user.name).must_equal(USER_NAME)
+      _(request.user.type).must_equal(USER_TYPE)
+    end
+
+    it 'handles nested arrays of both messages and hashes' do
+      user_hash = {
+        name: USER_NAME,
+        type: USER_TYPE,
+        posts: [
+          { text: POST_TEXT },
+          Google::Protobuf::Post.new(text: POST_TEXT)
+        ]
+      }
+      user = Google::Gax.to_proto(user_hash, Google::Protobuf::User)
+      _(user).must_be_kind_of(Google::Protobuf::User)
+      _(user.name).must_equal(USER_NAME)
+      _(user.type).must_equal(USER_TYPE)
+      _(user.posts).must_be_kind_of(Google::Protobuf::RepeatedField)
+      user.posts.each do |post|
+        _(post).must_be_kind_of(Google::Protobuf::Post)
+        _(post.text).must_equal(POST_TEXT)
+      end
+    end
+
+    it 'handles maps' do
+      request_hash = {
+        name: USER_NAME,
+        map_field: MAP
+      }
+      user = Google::Gax.to_proto(request_hash, Google::Protobuf::User)
+      _(user).must_be_kind_of(Google::Protobuf::User)
+      _(user.name).must_equal(USER_NAME)
+      _(user.map_field).must_be_kind_of(Google::Protobuf::Map)
+      user.map_field.each do |k, v|
+        _(MAP[k]).must_equal v
+      end
+    end
+
+    it 'handles IO instances' do
+      file = File.new('spec/fixtures/fixture_file.txt')
```

Comment:
Can `spec/fixtures/fixture_file.txt` be copied to the test dir in this PR to prepare for deleting spec in a follow-up PR?

Reply:
Moved.